index.php

```diff
@@ -0,0 +1,152 @@
+<?php
+ini_set('display_errors', 1);
+ini_set('display_startup_errors', 1);
+
+require __DIR__ . '/vendor/autoload.php';
+
+use Monolog\Logger;
+use Monolog\Handler\StreamHandler;
+use JoliCode\Slack\ClientFactory;
+use JoliCode\Slack\Exception\SlackErrorResponse;
+require "agenda.php";
+require "security.php";
+
+$log = new Logger('SlackApp');
+$log->pushHandler(new StreamHandler('access.log', Logger::DEBUG));
+
+//
+// Checking credentials
+// 
+if(!file_exists('credentials.json')) {
+    $log->error('credentials.json not found');    
+}
+
+$credentials_file_content = file_get_contents('./credentials.json');
+$credentials = json_decode($credentials_file_content);
+
+if(is_null($credentials)) {
+    $log->error('credentials.json is not json formated');
+    exit();
+}
+
+if(!property_exists($credentials, 'signing_secret') || !property_exists($credentials, 'slack_bot_token')) {
+    $log->error('signing_secret and/or slack_bot_token not present in credentials.json');
+    exit();
+}
+
+if(
+    !property_exists($credentials, 'caldav_url') ||
+    !property_exists($credentials, 'caldav_username') ||
+    !property_exists($credentials, 'caldav_password')) {
+    $log->error('Caldav credentials not present in credentials.json');
+    exit();
+}
+
+//
+// Extract request parts + HMAC check
+//
+
+$header = apache_request_headers();
```

Comment:
Will that work if this code runs on a non apache server (eg: nginx)? (Could it make sense to consider using $_SERVER instead?)

Reply:
Hop !  846dc58